internal/message.go

```diff
@@ -0,0 +1,177 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package internal
+
+import (
+	"fmt"
+	"net"
+
+	spoe "github.com/criteo/haproxy-spoe-go"
+)
+
+type message struct {
```

Comment:
I'd rather do something functions to get values from the map as string/int/ip.

```
func getString(args map[string]interface{}, key string) (string, error) {
   ...
}
```

Adding new struct to fetch values from a map looks overkill unless there are really important logic.

cc @anuraaga